src/nodes/number.cpp

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2025 Team Dissolve and contributors
+
+#include "nodes/number.h"
+#include <string>
+
+Number::Number(const std::variant<int, double> &value) : value_(value) {}
+
+Number::Number(int value) : value_(value) {}
+
+Number::Number(double value) : value_(value) {}
+
+Number &Number::operator=(const Number &other)
+{
+    value_ = other.value_;
+    return *this;
+}
+
+struct AddNumberVisitor
+{
+    template <class T1, class T2> auto operator()(T1 a, T2 b) const -> Number { return {a + b}; }
+};
+Number Number::operator+(const Number &other) const { return std::visit(AddNumberVisitor{}, value_, other.value_); }
```

Comment:
I'm a little curious whywe need a full class here, instead of just a lambda?  I would have thought that we could get away with:

```suggestion
Number Number::operator+(const Number &other) const { return std::visit([](auto a, auto b) -> Number {return {a+b};}, value_, other.value_); }
```

Is there an advanage to making the visitor class or some other context I've missed here that prevents the lambda from working?  These changes would need to apply to the other operator classes as well, if it's possible.

Reply:
Nope. Apparently I overcomplicated things here!  The explicit reference on the paraneter `a` _is_ required for assignment operators though.